Lesson32/Kadir/my-app/src/components/RecipeList/index.jsx

```diff
@@ -1,13 +1,39 @@
-export const RecipeList = ({recipes}) => {
+import "./styles.css";
+import { useState } from "react";
 
-    return(
-        <div>
-            <h2>Search Results</h2>
-            <ul>
-              {
-             recipes.map(recipe => <li key={recipe.idMeal} >{recipe.strMeal}</li>)   
+export const RecipeList = ({ recipes, handleRecipeSelect, selectedRecipe }) => {
+  const [isActive, setIsActive] = useState(true);
+
+  const toggleClick = () => {
+    setIsActive(!isActive);
+  };
+
+  return (
+    <div className="recipe-container">
+      <h2>Search Results</h2>
+      <ul className="recipe-list">
+        {recipes.map((recipe) => (
+          <li
+            key={recipe.idMeal}
+            onClick={() => {
```

Comment:
Right now the logic is a not working very well when de-selecting the recipe. I'd change it as following:
1. change this onClick to `onClick={() => handleClick(recipe)}`
2. Add handleClick function:
```
  const handleClick = (recipe) => {
// if there's a selected recipe, and if the selected recipe is the same that a user clicked on the last, clear selection   
 if (selectedRecipe && (recipe.idMeal === selectedRecipe.idMeal)) {
      handleRecipeSelect(null);
// otherwise select the clicked recipe
    } else {
      handleRecipeSelect(recipe);
    }
  }
```